### DISCUSSION TITLE
What sample rate does PE process my PipeWire clients at?

### DISCUSSION THREAD

**Discussion by gustafla** [wwmm/easyeffects#930] [Q&A] [answered]
Hello! I love this application, it's truly phenomenal. I just want to know, as an user of [AutoEQ](https://github.com/jaakkopasanen/AutoEq) correction, are the sample rate numbers in the top bar accurate? So even if the applications-view shows that 44100 is coming in, all DSP happens at 48000 as indicated in the top bar? I have nothing against PipeWire's resampling policy, in fact, it just simplifies using the Convolver to correct my headphones. I'm just getting paranoid if I'm selecting the right convolution wav file, because there is one for 44100 and another for 48000.

**Comment by wwmm** [comment, marked as answer]
>all DSP happens at 48000 as indicated in the top bar?

Yes. We set the rate of our effects pipeline to the same value PipeWire reports in our virtual sink. And as PipeWire does not update its rate on the fly yet this value will be used all the time.

>if I'm selecting the right convolution wav file

If necessary our convolver resamples the impulse file to the rate used in our effects pipeline. But if you have a 48 kHz impulse file you can save some processing time by loading it instead.

**Reply by gustafla** [reply]
Thanks! That's really nice. I'm not a signal processing expert and the maths on Wikipedia about FIR get too hard for me to follow. I'm probably being paranoid about audio quality, but zita-convolver does seem to use some kind of FFT as an optimization. Are the results how optimal for quality in addition to performance? Like is it going from time domain to frequency domain and then back again? Of course it sounds fine, but still a tiny bit different to the 9-band IIR EQ computed from the same data.

**Comment by wwmm** [comment]
>I'm not a signal processing expert

Although I am a physicist digital signal processing is not my original field. So I am not an expert too. Last time I tried to take a look at zita source code I did not understand a single thing XD. But as far as I remember zita does some kind of partitioned FFT with non uniform sizes tuned for realtime analysis. I never did any quality measurement. But so far I did not notice problems.
